Add cygwin support

plicease opened the pull request:
Cygwin has a /proc just like Linux (it is based on the Linux version in fact).

The first commit Devel::CheckOS could probably be trimmed a little bit, which is why I left it as a separate commit.  I can trim it if you prefer.

Thanks!

plicease committed:
Update Devel::CheckOS to 1.73, for Cygwin module

plicease committed:
Add support for cygwin which has a /proc based on Linux

pioto commented:
Thanks for the patch, but can you please tweak it?

It seems that Module::Build doesn't support the array notation you're using for license, at least not in version 0.3800, which my Cygwin install has at the moment:

```
$ git rev-parse HEAD
746ed996454b8d61f1c2d086f700caee15d0b81f

$ perl Build.PL
cygwin warning:
  MS-DOS style path detected: \Users\Pioto
  Preferred POSIX equivalent is: /cygdrive/c/Users/Pioto
  CYGWIN environment variable option "nodosfilewarning" turns off this warning.
  Consult the user's guide for more details about POSIX paths:
    http://cygwin.com/cygwin-ug-net/using.html#using-pathnames
Can't find dist packages without a MANIFEST file
Run 'Build manifest' to generate one

WARNING: Possible missing or corrupt 'MANIFEST' file.
Nothing to enter for 'provides' field in metafile.
Could not get valid metadata. Error is: Invalid metadata structure. Errors: 'ARRAY(0x600070400)' for 'license' does not have a URL scheme (resources -> license) [Validation: 1.4]
 at /usr/lib/perl5/5.14/Module/Build/Base.pm line 4559

Could not create MYMETA files
Creating new 'Build' script for 'Unix-Uptime' version '0.3701'
```

plicease commented:
Sure I will investigate this tomorrow. For me it was failing without the array notation.

plicease commented:
This suggests that it should be a list:

http://search.cpan.org/~dagolden/CPAN-Meta-2.142060/lib/CPAN/Meta/Spec.pm#resources

and this is the error that I get with a more recent Module::Build (or something that it depends on perhaps):

```
dare% perl Build.PL
Can't find dist packages without a MANIFEST file
Run 'Build manifest' to generate one

WARNING: Possible missing or corrupt 'MANIFEST' file.
Nothing to enter for 'provides' field in metafile.
Could not get valid metadata. Error is: Invalid metadata structure. Errors: Expected a list structure (resources -> license) [Validation: 2]
 at /home/ollisg/.perlbrew/libs/perl-5.18.2c34@dev/lib/perl5/Module/Build/Base.pm line 4557.

Could not create MYMETA files
Creating new 'Build' script for 'Unix-Uptime' version '0.3701'
dare% which_pm Module::Build
Module::Build        0.4208     /home/ollisg/.perlbrew/libs/perl-5.18.2c34@d...
Module::Build        0.4003     /home/ollisg/perl5/perlbrew/perls/perl-5.18....
```

give me a sec and I will see if I can figure out how to get it to work with both.

plicease committed:
Module::Build 0.4200 compatability

work around for this commit in Module::Build:

https://github.com/Perl-Toolchain-Gang/Module-Build/commit/044356d0e9934eaf45a9016786b6efa524cdf6f7

the license resource isn't converted from scalar to list as it should be when going from spec version 1 to 2.
Since the license is already specified as perl in the meta URL links to the text of the license is unnecessary

plicease commented:
As mentioned in e57eecbdeabcceb66b7eadb8e0f8dd11921e8fa5 commit message, this commit for Module::Build for newer versions does not convert the license resource:

https://github.com/Perl-Toolchain-Gang/Module-Build/commit/044356d0e9934eaf45a9016786b6efa524cdf6f7

This is why it fails with a scalar on newer versions but with a list in older versions.  An alternative to this would be to require Module::Build 0.4200 instead of 0.28.  I can rewrite the last commit if that is the solution that you prefer.

plicease commented:
Hi @pioto any thoughts on my proposed workaround?

pioto commented:
I'm not totally sure I want to force a dependency upon a newer Module::Build.

It seems weird that they would completely break backwards compatibility in this way; are you sure there isn't some other solution? As an alternative, I'd also consider just omitting the license stuff from the configuration.

Also, from skimming those commits, it looks like maybe the newer Module::Build is trying to parse my stuff against v2 of a Meta spec, can we tell it "no, really, this is still v1" somehow?

plicease commented:
Actually it looks to me like they are trying to coerce some fields from v1 to v2, but not this one.  I am guessing that not that many modules specify a URL license? Given that this module is licensed under license same as Perl 5 I am not sure it is even necessary?  I am not a lawyer of course, I just implemented the workaround since I  assumed that you had done so for a good reason.

plicease commented:
Since you didn't seem keen on my previous work around I've adjusted it to simply remove the license resource.  I suggest adding a LICENSE file.

plicease commented:
Hi @pioto, any more thoughts on this.  It would be nice to be able to use this module on cygwin.

pioto commented:
Hi,

Sorry for the delay in getting back to you.

Yeah, I read through [v1.4 META spec](http://module-build.sourceforge.net/META-spec-v1.4.html#license),  the [v2.0 META spec](http://search.cpan.org/dist/CPAN-Meta/lib/CPAN/Meta/Spec.pm#license), and [Module::Build::API](http://search.cpan.org/dist/Module-Build/lib/Module/Build/API.pod#license), and I see what you're saying now (that this is already covered by the license directive at [line 14 of my Build.PL](https://github.com/pioto/Unix-Uptime/blob/7d121134f67850ae2c891d24030b98399c908620/Build.PL#L14)).

So I'll do some final testing, and hopefully get your change merged soon.

plicease commented:
How about a release to CPAN?

pioto commented:
Version 0.4000 should be hitting a mirror near you shortly.

Thanks.

plicease commented:
Verified that it installs on my cygwin correctly thanks!
